Add 2022-09-08 agenda and minutes (#70)
solid/notifications-panel: meetings/2022-09-08.md

### Simplify flow diagrams
URL: https://github.com/solid/notifications/pull/108

> * AC: Authorization channel is out of scope, so it doesn't matter in this spec. I took the common parts and made that into one diagram. COuld improve with some relocation of 'Subscriber' in the diagram. The second diagram is to establish the connection for "Connected" client. The last one "Offline" clients and have two versions. Basically have a subscriber, independently of that have target/source happening.

Reviewer: ```suggestion
* AC: Authorization channel is out of scope, so it doesn't matter in this spec. I took the common parts and made that into one diagram. Could improve with some relocation of `Subscriber` in the diagram. The second diagram is to establish the connection for `Connected` client. The last one has `Offline` clients, and has two versions. Basically have a subscriber; independently of that, have target/source happening.
```

* RG: +1 on notification source/target. I would leave the connected as is. maybe some gap.

> * SC: +1 on Connected as is. Perhaps update to express resource-specific or storage-wide because right now it says "storage description resource".
> * SC: s/storage-description-resource/notification-channel-resource

Reviewer: There is no preceding `storage-description-resource`, though there is a preceding `"storage description resource"`, so maybe this?

```suggestion
* SC: +1 on Connected as is. Perhaps update to express resource-specific or storage-wide because right now it says "notification channel resource".
```

Author: The replacement suggestion was just a shorthand reference to the diagram (that AC was showing) and not the content of the minutes.

```suggestion
* SC: +1 on Connected as is. Perhaps update to express resource-specific or storage-wide because right now it says "storage description resource".
* SC: Suggest chanting "storage description resource" in the diagram to "notification channel resource"
```

* AC: yea, and step 2 description will change as well.

ACTION: AC to do minor updates.

### Discovery of Notification Subscription Metadata
URL: https://github.com/solid/notifications/issues/58

* SC: Notification Channel Resource is the current term for the resource that includes some information about the notification channel/service and its capabilities.

> * SC: Some people find `http://www.w3.org/ns/solid#storageDescription` useful - at least on the Solid Protocol level. Current ED explains how to take advantage of it.. in addition to `describedby`. Thoughts?

Reviewer: ```suggestion
* SC: Some people find `http://www.w3.org/ns/solid#storageDescription` useful — at least on the Solid Protocol level. Current ED explains how to take advantage of it, in addition to `describedby`. Thoughts?
```

* RG: Is it two different mechanisms available all the time?
* SC: Same same different. resource-specific or storage-wide cases.
* RG: But in the end it links to two places to get that information.

> * AC: Asking for clarity: in the link header, is all of the information in the header available or ?
> * RG: No. Will the info be in describedby or the storageDescripiton. How would they know about the storage description when they access the first resource.

Reviewer: ```suggestion
* AC: Asking for clarity: in the `link` header, is all of the information in the header available or not?
* RG: No. Will the info be in `describedby` or the `storageDescription`? How would they know about the storage description when they access the first resource?
```

* SC: examples:

```http
HEAD /foo

Link: </notification-resource>; rel="describedby"
Link: </notification-resource>; rel="http://www.w3.org/ns/solid#storageDescription"

> ```

Reviewer: Lines 69 thru 74 should be indented 2 space chars, such that they render as "inside" SC's comment (line 67).  GitHub won't let me suggest this change, because it doesn't handle embedded code fences properly.

Author: Resolved as per https://github.com/solid/notifications-panel/pull/70/commits/a40f782b82454a45238b5554adb674998bf5d371

* RG: Can we have a non-normative section describing guidance or that they can even split this information.
* SC: Perhaps. Right now just with terminology around the intention i.e., resource-specific or storage-wide.
* SC: https://solid.github.io/notifications/protocol#server-link-resource-notification-channel :
>When a server wants to enable applications to discover Notification Channels available to a **given resource**, the server MUST advertise the associated resources describing the channel information by responding to an HTTP request including a Link header with the rel value of describedby [POWDER-DR] and the Description Resource as link target [RFC8288].
* SC: https://solid.github.io/notifications/protocol#server-link-storage-notification-channel :
>When a server wants to enable applications to discover Notification Channels available to a **storage** in which a given resource is in, the server MUST advertise the associated resources describing the channel information by responding to an HTTP request including a Link header with the rel value of http://www.w3.org/ns/solid#storageDescription [SOLID-PROTOCOL] and the Description Resource as link target [RFC8288].

> * AC: For the sake of interoperability, I the notifications protocol, if we required describedby, and effectively said, the server could expose all the other headers but then maybe we don't need to say that. The two different mechanisms would be available if used specific against Solid Protocol server. But describedby would me more widely and generally understandable.

Reviewer: ```suggestion
* AC: For the sake of interoperability, I _______ the notifications protocol, if we required `describedby`, and effectively said, the server _could_ expose all the other headers but then maybe we don't need to say that. The two different mechanisms would be available if used specifically against a Solid Protocol server. But `describedby` would be more widely and generally understandable.
```

Author: ```suggestion
* AC: For the sake of interoperability, in the notifications protocol, if we required describedby, and effectively said, the server could expose all the other headers but then maybe we don't need to say that. The two different mechanisms would be available if used specific against Solid Protocol server. But describedby would me more widely and generally understandable.
```

> >When a server wants to enable applications to discover Notification Channels available to a **given resource**, the server MUST advertise the associated resources describing the channel information by responding to an HTTP request including a Link header with the rel value of describedby [POWDER-DR] and the Description Resource as link target [RFC8288].
> * SC: https://solid.github.io/notifications/protocol#server-link-storage-notification-channel :
> >When a server wants to enable applications to discover Notification Channels available to a **storage** in which a given resource is in, the server MUST advertise the associated resources describing the channel information by responding to an HTTP request including a Link header with the rel value of http://www.w3.org/ns/solid#storageDescription [SOLID-PROTOCOL] and the Description Resource as link target [RFC8288].
> * AC: For the sake of interoperability, I the notifications protocol, if we required describedby, and effectively said, the server could expose all the other headers but then maybe we don't need to say that. The two different mechanisms would be available if used specific against Solid Protocol server. But describedby would me more widely and generally understandable.

Reviewer: The blank lines are necessary to end the MD quotations, and the 2-space indents are required to "embed" the quotations in the SC bullets
```suggestion
  >When a server wants to enable applications to discover Notification Channels available to a **given resource**, the server MUST advertise the associated resources describing the channel information by responding to an HTTP request including a Link header with the rel value of describedby [POWDER-DR] and the Description Resource as link target [RFC8288].

* SC: https://solid.github.io/notifications/protocol#server-link-storage-notification-channel :
  >When a server wants to enable applications to discover Notification Channels available to a **storage** in which a given resource is stored, the server MUST advertise the associated resources describing the channel information by responding to an HTTP request including a `Link` header with the `rel` value of `http://www.w3.org/ns/solid#storageDescription` [SOLID-PROTOCOL] and the Description Resource as link target [RFC8288].

* AC: For the sake of interoperability, I the notifications protocol, if we required describedby, and effectively said, the server could expose all the other headers but then maybe we don't need to say that. The two different mechanisms would be available if used specific against Solid Protocol server. But describedby would me more widely and generally understandable.
```

* SC: Do you think the current text meets that?

> * AC: If I'm understanding this correctly, if you want to use --- funny requirement --- then you must do it this way. That's fine. Usually in other specs, if you want to enable, then you need to do x, y, z. Here we are saying If you want to do a, then do a, seems a little odd. One thing I can suggestion.. when would a server not want to enable this. I think the first one is a must. For the second one related to storage, that's where I think the current text is probably good. It doesn't make a hard requirement, but this is how you do it.

Reviewer: ```suggestion
* AC: If I'm understanding this correctly, if you want to use ______ — funny requirement — then you must do it this way. That's fine. Usually in other specs, if you want to enable _a_, then you need to do _x, y, z_. Here, we are saying, "If you want to do _a_, then do _a_." Seems a little odd. One thing I can suggest: when would a server not want to enable this? I think the first one is a must. For the second one related to storage, that's where I think the current text is probably good. It doesn't make a hard requirement, but this is how you do it.
```

Author: ```suggestion
* AC: If I'm understanding this correctly, if you want to use this - a funny requirement - then you must do it this way. That's fine. Usually in other specs, if you want to enable, then you need to do x, y, z. Here we are saying If you want to do a, then do a, seems a little odd. One thing I can suggestion.. when would a server not want to enable this. I think the first one is a must. For the second one related to storage, that's where I think the current text is probably good. It doesn't make a hard requirement, but this is how you do it.
```

> * SC: Could we update the requirement for storageDescription?

Reviewer: ```suggestion
* SC: Could we update the requirement for `storageDescription`?
```

> * RG: If I'm accessing a resource and get describedby, then that's extra info that I can follow that.

Reviewer: ```suggestion
* RG: If I'm accessing a resource and get `describedby`, then that's extra info that I can follow.
```

> * RG: Suppose there isn't a describedby, and only storageDescription, I would think that would be applicable for all the other resources on the same storage. only when there is a deviation, that'd when I would expect a describedby. but when there is a describedby, I would get that. there would be an extra fetch any way.

Reviewer: ```suggestion
* RG: Suppose there isn't a `describedby`, and only `storageDescription`, I would think that would be applicable for all the other resources on the same storage. I would expect a `describedby` only when there is a deviation. But when there is a `describedby`, I would get that. There would be an extra fetch anyway.
```

> * SC: so with my dev hat on: if I see describedby, I would get it and if I find notifications channel info, good enough for me. I stop. if I don't see the link header or don't find the info I want, then ill look into storage description link if there is one.. and follow to its description and so on.

Reviewer: ```suggestion
* SC: so with my dev hat on: if I see `describedby`, I would get it, and if I find notifications channel info, good enough for me; I stop. If I don't see the `link` header or don't find the info I want, then I'll look into storage description link if there is one, and follow to its description, and so on.
```

* SC: That may cover majority of implementations but I don't have data to back that up.
* RG: if described by why would there be storage description

> * AC: say generic term like `rdfs:label` and put some info there but then you could also have more specific property like `skos:prefLabel` which might have the same info. but what this means is that you don't know what clients may be interacting.. some clients may not know SKOS but then they'd use the RDFS label. But other clients may like the more precise term. that puts extra burden on server but making the interop more widely available.

Reviewer: ```suggestion
* AC: Say generic term like `rdfs:label` and put some info there, but then you could also have more specific property like `skos:prefLabel`, which might have the same info. But what this means is that you don't know what clients may be interacting; some clients may not know SKOS but they might use the RDFS label. Other clients may like the more precise term. That puts extra burden on server, but makes the interop more widely available.
```

> * SC: Proposal:

Reviewer: ```suggestion
```

PROPOSAL: Keep the requirement as is. We can drop that when we get some implementation feedback from clients.

* AC: +1
* RG: +1

> * RG: AC, you are saying that storage description would have more details data. isn't that more of a generic thing to do.
> * AC: I'm saying the semantics of storage description (relation) is more precise than describedby.

Reviewer: ```suggestion
* RG: AC, you are saying that storage description would have more detailed data. Isn't that more of a generic thing to do?
* AC: I'm saying the semantics of storage description (relation) is more precise than `describedby`.
```

* RG: I would expect more specific semantics for a resource.

> * SC: so instead of describedby, more like `notify:notificationResource` ?
> * AC: I think the idea is that the describedby is going to be specific to a resource, and storage description is for the entire storage. there are two possibilities. the information follows the links. the other is that thye are different. in the case where they are different, and in the case with describedby, it would have more information but that would up to the implementation in the end.

Reviewer: ```suggestion
* SC: so instead of `describedby`, more like `notify:notificationResource`?
* AC: I think the idea is that the `describedby` is going to be specific to a resource, and storage description is for the entire storage. There are two possibilities. The information follows the links. The other is that they are different. In the case where they are different, and in the case with `describedby`, it would have more information, but that would be up to the implementation, in the end.
```

* SC: similar to superset from earlier (storage + resource specific notification channel)
* RG: ???
* AC: A client may decide to fetch only one. A client may have different reasons to treat them differently. I think it is up to the client.

> * RG: But the client to make that information, ??? thought about how a 

Reviewer: indecipherable
```suggestion
* RG: But the client to make that information, ??? thought about how a 
```

Author: @CxRes , pardon me for not capturing this properly. Feel free to make a suggestion if you recall the flow of the conversation - same goes for line 108. Okay to leave as is or omit line too.

Reviewer 2: Is it OK if I very slightly embellish the conversation to create a better structure (I don't intend to change the purport of the conversation though)?

* RG: If there is a less specific first and more general thing later, if I have a storage description then that's good enough then I would do that.
* SC: ...
* RG: ...

> * AC: Maybe a way to get around this, in the spec, there are two mechanisms in discovery, then at least one must be present. for storage-wide, use x, for resource specific, use y. what happens is, if only one of those is present, then a client follows that, if both present and pointing to different URLs, then client will decide on info that's specific to that resource or the storage as a whole.

Reviewer: ```suggestion
* AC: Maybe a way to get around this: in the spec, there are two mechanisms in discovery, and at least one must be present. For storage-wide, use _x_; for resource specific, use _y_. What happens is, if only one of those is present, then a client follows that; if both are present and pointing to different URLs, then client will decide on info that's specific to that resource or to the storage as a whole.
```
